Info collection

tclhh opened the pull request:
作业说明:
- 本次作业主要介绍信息收集的一些手段，并没有整理收集的信息。只是作为作业的第一阶段，为后面的工作的展开做铺垫。

作业问题:
- 关于工具的分类问题，可能写的比较模糊，特别是搜索引擎 和 枚举 ，KALI中的很多工具能用 搜索引擎，又能用暴力穷举的功能。
- 可能部分工具的使用写的并不是很详细，可能会有一些问题。
- 原文还有许多问题需要解答

tclhh committed:
IVRE 搭建实验

tclhh committed:
提交dns枚举的工具及方法

tclhh committed:
修改相对路径问题

作业说明:
- 本次作业主要介绍信息收集的一些手段，并没有整理收集的信息。只是作为作业的第一阶段，为后面的工作的展开做铺垫。

作业问题:
- 关于工具的分类问题，可能写的比较模糊，特别是搜索引擎 和 枚举 ，KALI中的很多工具能用 搜索引擎，又能用暴力穷举的功能。
- 可能部分工具的使用写的并不是很详细，可能会有一些问题。
- 原文还有许多问题需要解答。

tclhh committed:
修改排版问题

tclhh committed:
修改遗漏的相对路径

c4pr1c3 commented:
* PR 存在冲突，这不是个好习惯，需要解决掉这个冲突；
* `本地的 hosts 文件没有这个域名的缓存记录`？ hosts 文件里是「静态配置」的域名解析记录，不是「缓存」；
* [整段的复制粘贴](https://blog.csdn.net/homedesk/article/details/77989241) 就不要留在实验报告里了，没有意义，还耽误我批改作业；
* `主题` 你不知道是啥意思？X.509 格式证书里的「Subject」知道是干嘛的吗？不知道就去 Google 一下。你这里截图上的`主题`就是「Subject」的翻译而已。表格里第一列实际上只是 Subject 字段集合里的 CN 字段；
* 至于「DNS 名称数量」？多说无益，看具体例子： https://www.qq.com 和对应的 [Google 透明度报告](https://transparencyreport.google.com/https/certificates/sBxT6Pp%2BEpRQRg7t5BOsDTB3uINF2OcjvX65k19KWMU%3D) 对 「DNS 名称」是不是有了直观认识了？这是 X.509 的一个扩展字段（主题备用名称）：X509v3 Subject Alternative Name

![transparencyreport-1](https://user-images.githubusercontent.com/5326304/50423134-553cc380-088c-11e9-85a1-5e69bff9d26e.png)

![transparencyreport-2](https://user-images.githubusercontent.com/5326304/50423136-55d55a00-088c-11e9-98ae-d677a505714a.png)

![transparencyreport-3](https://user-images.githubusercontent.com/5326304/50423135-55d55a00-088c-11e9-85ae-73805df47c7b.png)

* 以上，「主题」和「DNS 名称」都是在签发证书的时候由证书的申请者在生成证书时配置好的，用于标识该证书可以用于证明哪些域名是可以用该证书进行「真实性」证明；
* `Censys.io` 你给的截图不是 qq.com 的子域名，只是域名中包含 `qq.com` 关键字而已；
* DNS 域传送是一种「配置缺陷」利用手段，能够查询到指定域名的所有子域名信息的前提是域名服务器管理员配置不当；
* 实验报告如果把你用到的这一堆方法，按照「查询原理」做一下「归类」就更好了，而不是像现在的报告这样，「一个接一个」方法的简单重复实验，特别是对于`「基于字典」的子域名枚举` 这一种方法你用 N 种「工具」去做实验，其实意义不如你「每一种方法」用一种工具实验，尝试更多「方法」，而不是像现在这样尝试更多同类型「工具」；

tclhh commented:
- 之前的 IVRE 也是 readme.md 命名冲突 ，之后我会一起解决一下。

- `本地的 hosts 文件没有这个域名的缓存记录` 这段我是 copy 的 ，hosts 文件是静态配置的(之前我也看过...)

- 一般我在 `BASIS` 中会有大段的引用 , 基本都是复制粘贴的，但是都是我之前不懂的地方，所以才要复制粘贴的。关于原文，我也在 `参阅` 中给出了链接。`BASIS` 之外如果出现分点概括的情况，那些基本上是我自己概括的，一般这些地方很可能会出现遗漏或错误，才是真正需要注意的地方。

- `Subject` 没想起来...... 以前学过的东西都忘了...... `主题` 内容是 `CN字段` 涨姿势了。

- `DNS 名称数量` 果然直接看证书才是正解！！！👍 不过在之后的 `crt.sh` 的截图的证书中也有这个字段，不过当时没有发现.....

- `Cersys.io`  截图确实不是 qq.com 子域名，是我在 qq.com 子域名的页面中随便点开的一个页面。

- DNS 域传送这一块我在原文中提到了`查阅资料知 , 网络管理员禁止服务器接收来自未认证DNS服务器的AXFR请求 , 会导致 AXFR 请求失败`， 所以才得出了 `AXFR 得到的信息比起之前的方法少了很多, 在该应用场景下并不是很合适` 的结论，一般 DNS 域传输还是可以收集到很多信息的。


- 分类问题我在交作业之前也提到了 `关于工具的分类问题，可能写的比较模糊，特别是搜索引擎 和 枚举 ，KALI中的很多工具能用 搜索引擎，又能用暴力穷举的功能` ，当然也有一些网站的原理我没有理解，所以没有分类，比如: Virustotal 等。是因为我的资料收集不足的原因，所以增加了老师批阅时的负担，之后我会注意这些问题的，同时减少篇幅。而且这个作业只是一个工具总结，真正的实验是会结合所有合适的工具的实验结果的。


- 最后，在实验报告中还有一些问题没有解决，这里列举一下:
   - `theharvester` `-b` 参数指定使用 `bing ` 为什么没有任何结果？
   - `透明度报告`的遗漏情况总结是否有错？ 
    ![default](https://user-images.githubusercontent.com/26518517/50424408-f5e7af00-089d-11e9-83dd-2cd7ae9adf80.png)
   - `dnsenum` 为什么会报错？这个我查了很久，但是没有解决......
   ![default](https://user-images.githubusercontent.com/26518517/50424430-67bff880-089e-11e9-989d-1ffd173f72a8.png)

c4pr1c3 commented:
> `theharvester` `-b` 参数指定使用 `bing ` 为什么没有任何结果？

守着源代码，就不要等靠要问我了，调试见真章。

> `透明度报告`的遗漏情况总结是否有错？ 

肯定会有遗漏啊，不用 HTTPS 的域名、用了 HTTPS 但没有启用透明度报告的域名就肯定不会被收录呀。

> `dnsenum` 为什么会报错？这个我查了很久，但是没有解决......

守着源代码，就不要等靠要问我了，调试见真章。

tclhh committed:
Delete README.md

tclhh committed:
Create README.md

tclhh committed:
少许修改

- 增加了 ESD 枚举, 并上传结果
- 本来打算使用 Rapid7 的数据集 , 直接 grep 筛选 , 结果看了一下压缩包有 15 个G，最后没有下载......（官网的github Forward DNS中提供 curl 的方法,尝试了一下,发现不行...）

tclhh committed:
上传 ESD 枚举结果

Wzy-CC committed:
新增了Google Hacking和百度信息收集

增加了百度信息收集md和谷歌收集md

Wzy-CC committed:
排版格式更改

更改了图片的相对路径

Wzy-CC committed:
排版更改

一张图片路径更改
